contracts/utils/Strings.sol

```diff
@@ -51,17 +51,18 @@ library Strings {
             string memory buffer = new string(length);
             /// @solidity memory-safe-assembly
             assembly {
-                let pos := add(buffer, 32)
-                let ptr := add(pos, length)
-
+                let ptr := add(buffer, add(32, length))
                 for {
 
-                } gt(ptr, pos) {
+                } 1 {
 
                 } {
                     ptr := sub(ptr, 1)
-                    mstore8(ptr, add(48, mod(value, 10)))
+                    mstore8(ptr, byte(mod(value, 10), _SYMBOLS))
                     value := div(value, 10)
+                    if iszero(value) {
```

Comment:
No, it's equivalent, because it's performed once per iteration anyway. We need to cover the `0` case, so we needed to turn a "while" loop into a "do-while". If we special-case it, we can move `iszero` to the for loop exit clause.

Reply:
I agree that the isZero must be at the end so that `toString(0)` is correctly processed.